src/backend/InvenTree/InvenTree/models.py

```diff
@@ -78,6 +78,81 @@ def has_field_changed(self, field_name):
         return field_name in self.get_field_deltas()
 
 
+class StatusCodeMixin:
```

Comment:
I think this would better fit into the generic / states part of the code base to ensure we provide proper decoupling in tests and code